src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp

```diff
@@ -21,6 +21,138 @@ constexpr char biosIntType[] =
 constexpr char biosEnumType[] =
     "xyz.openbmc_project.BIOSConfig.Manager.AttributeType.Enumeration";
 
+std::string HypEthInterface::getAttrFromBiosTable(std::string attrName)
+{
+    constexpr auto BIOS_SERVICE = "xyz.openbmc_project.BIOSConfigManager";
+    constexpr auto BIOS_OBJPATH = "/xyz/openbmc_project/bios_config/manager";
+    constexpr auto BIOS_MGR_INTF = "xyz.openbmc_project.BIOSConfig.Manager";
+
+    try
+    {
+        using getAttrRetType =
+            std::tuple<std::string, std::variant<std::string, int64_t>,
+                       std::variant<std::string, int64_t>>;
+        getAttrRetType ip;
+        auto method = bus.new_method_call(BIOS_SERVICE, BIOS_OBJPATH,
+                                          BIOS_MGR_INTF, "GetAttribute");
+
+        method.append(attrName);
+
+        auto reply = bus.call(method);
+
+        std::string type;
+        std::variant<std::string, int64_t> currValue;
+        std::variant<std::string, int64_t> defValue;
+        reply.read(type, currValue, defValue);
+        return std::get<std::string>(currValue);
+    }
+    catch (const sdbusplus::exception::SdBusError& ex)
+    {
+        log<level::ERR>("Failed to get the hostname from bios table",
+                        entry("ERR=%s", ex.what()));
+    }
+    return "";
+}
+
+void HypEthInterface::watchBaseBiosTable()
+{
+    auto BIOSAttrUpdate = [&](sdbusplus::message::message& m) {
+        std::map<std::string, std::variant<BiosBaseTableType>>
+            interfacesProperties;
+
+        std::string objName;
+        m.read(objName, interfacesProperties);
+
+        // Check if the property change signal is for BaseBIOSTable property
+        // If found, proceed; else, continue to listen
+        auto find = interfacesProperties.find("BaseBIOSTable");
+        if (find == interfacesProperties.end())
+        {
+            // Return & continue to listen
+            return;
+        }
```

Comment:
this can be replaced with c++ 17 map.contains("key").

Reply:
Done